keps/808-GangConfig/README.md

```diff
@@ -0,0 +1,444 @@
+
+The API below would fit my purpose:
+
+```
+apiVersion: jobset.x-k8s.io/v1
+kind: JobSet
+metadata:
+  name: sample-jobset
+spec:
+    replicatedJobs:
+    - name: replicated-job-1
+      replicas: 2
+      scheduleAtOnce: true
+      template:
+        spec:
+          parallelism: 4
+          completions: 4
+          template:
+            spec:
+              containers:
+                - name: initializer
+                  image: busybox
+                  command:
+                    - /bin/sh
+                    - -c
+                    - "echo 'initializer runs for 10 seconds' && sleep 10"
+    - name: replicated-job-2
+      replicas: 1
+      scheduleAtOnce: true
+      dependsOn:
+        - name: replicated-job-1
+          status: Complete
+      template:
+        spec:
+          parallelism: 3
+          completions: 3
+          template:
+            spec:
+              containers:
+                - name: launcher
+                  image: busybox
+                  command:
+                    - /bin/sh
+                    - -c
+                    - "echo 'launcher runs for 20 seconds' && sleep 20"
+                  startupProbe:
+                    exec:
+                      command:
+                        - echo
+                        - "started"
+                    initialDelaySeconds: 5
+```
+
+#### Story 3
+
+As a user, I want to schedule groups of parallel workers in my workload to start at once i.e. all pods in a Job replica should be scheduled at once or not at all.
+
+The API below would fit my purpose:
+
+```
+apiVersion: jobset.x-k8s.io/v1
+kind: JobSet
+metadata:
+  name: sample-jobset
+spec:
+  replicatedJobs:
+    - name: replicated-job-1
+      replicas: 2
+      scheduleAtOnce: true
+      scheduleAtOncePerReplica: true
+      template:
+        spec:
+          parallelism: 4
+          completions: 4
+          template:
+            spec:
+              containers:
+                - name: initializer
+                  image: busybox
+                  command:
+                    - /bin/sh
+                    - -c
+                    - "echo 'initializer runs for 10 seconds' && sleep 10"
+    - name: replicated-job-2
+      replicas: 3
+      scheduleAtOnce: true
+      scheduleAtOncePerReplica: true
+      template:
+        spec:
+          parallelism: 3
+          completions: 3
+          template:
+            spec:
+              containers:
+                - name: launcher
+                  image: busybox
+                  command:
+                    - /bin/sh
+                    - -c
+                    - "echo 'launcher runs for 20 seconds' && sleep 20"
+                  startupProbe:
+                    exec:
+                      command:
+                        - echo
+                        - "started"
+                    initialDelaySeconds: 5
+```
```

Comment:
Will this result in a Workload similar to the following ?

```yaml
apiVersion: scheduling.k8s.io/v1alpha1
kind: Workload
metadata:
  name: sample-jobset
spec:
  podGroups:
  - name: replicated-job-1
    replicas: 2
    policy:
      kind: Gang
      gang:
        minCount: 4
  - name: replicated-job-2
     policy:
     replicas: 3
      kind: Gang
      gang:
        minCount: 3
```

With pods on replicated-job-1 and replicated-job-2 pointing to their respective (and different) pod groups + a different PodGroupReplicaIndex for each replica ?

Reply:
It could help to better understand the mapping by adding `workload` and `pod` examples for all three use cases right before `### Implementation` so it's obvious how the corresponding mappings work. The pod manifest can be simplified by showing only the relevant fields. E.g. `.name`, `.spec.workload`